cpp/src/utilities/error_utils.hpp

```diff
@@ -25,6 +25,19 @@
 #define GDF_REQUIRE(F, S) \
   if (!(F)) return (S);
 
+/**---------------------------------------------------------------------------*
+ * @brief a version of GDF_REQUIRE for expressions of type `gdf_error` rather
+ * than booleans
+ *
+ * This macro is sort-of DEPRECATED.
+ *
+ *---------------------------------------------------------------------------**/
+#define GDF_TRY(_expression) do { \
```

Comment:
Do we still need `GDF_TRY` after adding `CUDF_TRY`?

Reply:
Yes. Just think of the top-level cuDF API functions - they can't just be made to drop returning `gdf_error`...